### ISSUE TITLE
How to Disable the 'Learn CodeWhisperer' Page from Appearing on VSCode Startup

### ISSUE THREAD

**Issue by LHR-lhr** [aws/aws-toolkit-vscode#3910]
## System details (run the `AWS: About Toolkit` command)

-   OS: Windows_NT x64 10.0.19045
-   Visual Studio Code version: 1.83.1
-   AWS Toolkit version: 1.95.0

## Question

Every time I launch Visual Studio Code (VSCode), a 'Learn CodeWhisperer' page automatically pops up. I am looking for a solution to disable this page from appearing on startup.

### Steps to Reproduce:

1、Launch Visual Studio Code.
2、Observe the automatic appearance of the 'Learn CodeWhisperer' page.

### Expected Behavior:
I would like to find a way to prevent the 'Learn CodeWhisperer' page from appearing every time I start VSCode.

![image](https://github.com/aws/aws-toolkit-vscode/assets/48155416/7624dbfb-2a3c-414c-b049-d9c6bacacd02)

**Comment by justinmk3** [comment]
This page should not show after the first start. That decision is remembered in vscode setting `aws.suppressPrompts`. 

In vscode settings, search for `aws.suppressPrompts` , what does it show? Is `codeWhispererNewWelcomeMessage` checked? 

<img width="652" alt="image" src="https://github.com/aws/aws-toolkit-vscode/assets/55561878/0c854bac-79c4-493c-8196-3722699e63fe">

Is it possible your settings.json file is getting overwritten by some process?

**Comment by LHR-lhr** [comment]
Thank you so much! This solved my issue! 

I couldn't find the `codeWhispererNewWelcomeMessage` setting when I searched for `aws.suppressPrompt`s in VSCode settings. 
![image](https://github.com/aws/aws-toolkit-vscode/assets/48155416/3546fc98-540c-4e53-a69e-46f22d07597b)

So, I added the following to my `settings.json`:
```
"aws.suppressPrompts": {
     "codeWhispererNewWelcomeMessage": true,
 }
```

**Comment by justinmk3** [comment]
That suggests something is going wrong with the vscode settings system. 

Can you share your logs for `Extension Host`:
![image](https://github.com/aws/aws-toolkit-vscode/assets/55561878/0e4ca110-953b-4f48-9b55-c4d01e92dacc)

and also the AWS Toolkit logs (check that private info is removed)? 

1. In VSCode settings, ensure that "aws log level" is "debug".
2. Perform the steps to reproduce the issue.
3. Use the `AWS: View Toolkit Logs` command to get the logs.

**Comment by LHR-lhr** [comment]
Oh! I've realized that I made a somewhat foolish mistake. 
When I tried to reproduce the issue, I found a syntax error in the last line of my `settings.json` file! After fixing this error, I couldn't reproduce the issue anymore. Upon reviewing the `settings.json` file, the configuration
```
"aws.suppressPrompts": {
     "codeWhispererNewWelcomeMessage": true,
 }
```
was successfully written. Therefore, I suspect that the previous issue was likely due to the syntax error in the last line of settings.json, which was causing the extension's settings not to be written correctly. 
Nevertheless, I want to express my gratitude for taking the time to answer my question.

**Comment by justinmk3** [comment]
> I found a syntax error in the last line of my `settings.json`

That's helpful, thanks for sharing that! We should probably check for this and show a useful message instead of silently failing.

I tried to reproduce the issue by creating a syntax error and vscode itself correctly detected it:
<img width="535" alt="image" src="https://github.com/aws/aws-toolkit-vscode/assets/55561878/867fc6e7-302d-4b80-9313-898720a26810">

Is it possible that you had enabled "Do not disturb" mode, so the notification was hidden?

Or could you give an example of the syntax error?

**Comment by LHR-lhr** [comment]
My setting.json file error is like this, vscode does not notify the
settings.json error
[image: 微信截图_20231023085421.png]

Justin M. Keyes ***@***.***> 于2023年10月20日周五 18:53写道：

**Comment by justinmk3** [comment]
Thanks! If you can try the prerelease from https://github.com/aws/aws-toolkit-vscode/releases/tag/prerelease , AWS Toolkit should detect this and warn about it.

**Comment by justinmk3** [comment]
Implemented in AWS Toolkit 1.96
